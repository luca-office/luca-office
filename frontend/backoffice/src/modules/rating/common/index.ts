export * from "./participant-rating-progress-bar/participant-rating-progress-bar"
export * from "./rater-invitation-overlay/rater-invitation-overlay"
export * from "./rater-paper/rater-paper"
export * from "./raters-card/raters-card"
export * from "./rating-action-button/rating-action-button"
export * from "./rating-detail-scenario-container/rating-detail-scenario-container"
