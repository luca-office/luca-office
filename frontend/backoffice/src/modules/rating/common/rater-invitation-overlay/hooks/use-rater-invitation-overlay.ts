import * as React from "react"
import {useCheckLogin, useInviteSurveyRaters} from "shared/graphql/hooks"
import {ratingsQuery} from "shared/graphql/queries"
import {exists} from "shared/utils"

export interface UseRaterInvitationOverlayHook {
  readonly dataLoading: boolean
  readonly actionLoading: boolean
  readonly inviteRaters: (emails: string[]) => Promise<void>
  readonly setEmails: (emails: string[]) => void
  readonly showRaterWarning: boolean
}

interface UseRaterInvitationOverlayParams {
  readonly surveyId: UUID
  readonly onDismiss: () => void
}

export const useRaterInvitationOverlay = ({
  surveyId,
  onDismiss
}: UseRaterInvitationOverlayParams): UseRaterInvitationOverlayHook => {
  const [emailAddresses, setEmailAddresses] = React.useState<string[]>([])

  const {account, checkLoginLoading: accountLoading} = useCheckLogin()
  const {inviteSurveyRaters, inviteSurveyRatersLoading} = useInviteSurveyRaters(surveyId, [
    {query: ratingsQuery, variables: {surveyId}}
  ])

  const showRaterWarning = account
    .map(({email}) => exists(emailAddress => emailAddress === email, emailAddresses))
    .getOrElse(false)

  const inviteRaters = (emails: string[]) =>
    new Promise<void>((resolve, reject) => {
      inviteSurveyRaters(emails)
        .then(() => {
          onDismiss()
          resolve()
        })
        .catch(reject)
    })

  const setEmails = (emails: string[]) => setEmailAddresses(emails)

  return {
    dataLoading: accountLoading,
    actionLoading: inviteSurveyRatersLoading,
    inviteRaters,
    setEmails,
    showRaterWarning
  }
}
