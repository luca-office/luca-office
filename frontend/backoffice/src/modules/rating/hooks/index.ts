export * from "./use-project-module-rating"
export * from "./use-rating"
