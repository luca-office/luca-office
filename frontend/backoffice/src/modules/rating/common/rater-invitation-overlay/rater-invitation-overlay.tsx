import * as React from "react"
import {Overlay} from "shared/components"
import {IconName} from "shared/enums"
import {useLucaTranslation} from "shared/translations"
import {ManageInvitationsModal} from "../../../common"
import {useRaterInvitationOverlay} from "./hooks/use-rater-invitation-overlay"

export interface RaterInvitationOverlayProps {
  readonly surveyId: UUID
  readonly existingRaterEmails: string[]
  readonly onDismiss: () => void
  readonly dataLoading?: boolean
}

export const RaterInvitationOverlay: React.FC<RaterInvitationOverlayProps> = ({
  surveyId,
  existingRaterEmails,
  onDismiss,
  dataLoading: parentDataLoading
}) => {
  const {t} = useLucaTranslation()

  const {actionLoading, inviteRaters, setEmails, showRaterWarning} = useRaterInvitationOverlay({
    surveyId,
    onDismiss
  })

  return (
    <Overlay>
      <ManageInvitationsModal
        labelKey={"rating__invite_title"}
        customInfoTextKey={"rating__invite_description"}
        inviteAttendeesTitleKey={"rating__invite_emails_title"}
        onDismiss={onDismiss}
        existingInvitations={existingRaterEmails}
        handleInvitations={inviteRaters}
        invitationProcessLoading={actionLoading}
        dataLoading={parentDataLoading}
        onEmailAddressesChange={setEmails}
        errorConfig={{icon: IconName.Alert, text: t("rating__invite_raters_warning")}}
        showError={showRaterWarning}
      />
    </Overlay>
  )
}
