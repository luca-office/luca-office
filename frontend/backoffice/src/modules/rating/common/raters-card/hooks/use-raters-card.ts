import {isEqual} from "lodash-es"
import * as React from "react"
import {useRatings, useSurveyUserAccounts} from "shared/graphql/hooks"
import {Rating, UserAccount} from "shared/models"
import {exists, find, isDefined} from "shared/utils"

interface RaterEntity {
  readonly raterId: UUID
  readonly email: string
  readonly finalized: boolean
}

export interface UseRatersCardHook {
  readonly ratersCount: number
  readonly raterEntities: RaterEntity[]
  readonly raterEmails: string[]
  readonly inviteOverlayVisible: boolean
  readonly showInviteOverlay: () => void
  readonly hideInviteOverlay: () => void
  readonly raters: UserAccount[]
  readonly ratings: Rating[]
  readonly isRatingFinalized: boolean
}

export const useRatersCard = (surveyId: UUID): UseRatersCardHook => {
  const ratersRef = React.useRef<UserAccount[]>([])
  const ratingsRef = React.useRef<Rating[]>([])

  const [inviteOverlayVisible, setInviteOverlayVisible] = React.useState(false)

  const {surveyUserAccounts: raters} = useSurveyUserAccounts(surveyId)
  const {ratings: allRatings} = useRatings(surveyId)

  if (!isEqual(ratersRef.current, raters)) {
    ratersRef.current = raters
  }

  const ratings = allRatings.filter(rating => !rating.isFinalScore)

  if (!isEqual(ratingsRef.current, ratings)) {
    ratingsRef.current = ratings
  }

  const isRatingFinalized = find(({isFinalScore}) => isFinalScore, allRatings).exists(({finalizedAt}) =>
    isDefined(finalizedAt)
  )
  const ratersCount = ratersRef.current.length
  const raterEntities = React.useMemo<RaterEntity[]>(
    () =>
      ratersRef.current.map(rater => ({
        raterId: rater.id,
        email: rater.email,
        finalized: exists(
          rating => !rating.isFinalScore && rating.userAccountId === rater.id && isDefined(rating.finalizedAt),
          ratingsRef.current
        )
      })),
    [ratersRef.current, ratingsRef.current]
  )
  const raterEmails = raterEntities.map(({email}) => email)

  const showInviteOverlay = () => setInviteOverlayVisible(true)
  const hideInviteOverlay = () => setInviteOverlayVisible(false)

  return {
    ratersCount,
    raterEntities,
    raterEmails,
    inviteOverlayVisible,
    showInviteOverlay,
    hideInviteOverlay,
    raters: ratersRef.current,
    ratings: ratingsRef.current,
    isRatingFinalized
  }
}
