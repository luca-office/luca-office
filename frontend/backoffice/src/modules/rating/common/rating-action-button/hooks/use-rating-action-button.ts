import * as React from "react"
import {useFinalizeRating, useRatings, useSurveyInvitations} from "shared/graphql/hooks"
import {every, find, isDefined} from "shared/utils"
import {useProjectModuleRating} from "../../../hooks"

export interface UseRatingActionButtonHook {
  readonly dataLoading: boolean
  readonly actionLoading: boolean
  readonly buttonDisabled: boolean
  readonly showOrly: () => void
  readonly onCancel: () => void
  readonly onConfirm: () => void
  readonly isOrlyVisible: boolean
  readonly finalRatingFinalized: boolean
}

export const useRatingActionButton = (surveyId: UUID): UseRatingActionButtonHook => {
  const [isOrlyVisible, setIsOrlyVisible] = React.useState(false)

  const {ratings: allRatings, ratingsLoading: allRatingsLoading} = useRatings(surveyId)
  const {finalizeRating, finalizeRatingLoading} = useFinalizeRating(surveyId)
  const {surveyInvitationsLoading, surveyInvitations} = useSurveyInvitations(surveyId)
  const {loading: allProjectModulesRatedDataLoading, areAllProjectModulesRated} = useProjectModuleRating(surveyId, true)

  const finalRating = find(({isFinalScore}) => isFinalScore, allRatings)

  const allParticipantsRated = surveyInvitations.length === 0 || every(areAllProjectModulesRated, surveyInvitations)
  const finalRatingFinalized = finalRating.exists(rating => isDefined(rating.finalizedAt))
  const canRatingBeFinalized = !finalRatingFinalized && allParticipantsRated

  const showOrly = () => setIsOrlyVisible(true)

  const hideOrly = () => setIsOrlyVisible(false)

  const onConfirm = () =>
    canRatingBeFinalized && finalRating.forEach(rating => finalizeRating(rating.id).then(() => hideOrly()))

  return {
    dataLoading: allRatingsLoading || allProjectModulesRatedDataLoading || surveyInvitationsLoading,
    actionLoading: finalizeRatingLoading,
    buttonDisabled: !canRatingBeFinalized,
    finalRatingFinalized,
    showOrly,
    onConfirm,
    onCancel: hideOrly,
    isOrlyVisible
  }
}
