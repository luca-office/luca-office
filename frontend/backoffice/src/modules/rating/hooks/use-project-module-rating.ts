import {useApolloClient} from "@apollo/client"
import {isEqual} from "lodash-es"
import * as React from "react"
import {
  useCodingCriteriaByItemsList,
  useFreetextQuestionRatingsByRatingsList,
  useScenarioCodingItemRatingsByRatingsList
} from "shared/components"
import {getCodingItemsFromCodingDimensions} from "shared/components/rating/utils"
import {ProjectModuleProgressType, ProjectModuleType} from "shared/graphql/generated/globalTypes"
import {useProjectModulesLazy, useRatings} from "shared/graphql/hooks"
import {useSurveyLight} from "shared/graphql/hooks/queries/survey/use-survey-light"
import {CodingDimension, CodingItem, ProjectModule, Rating, SurveyInvitationLight, SurveyLight} from "shared/models"
import {every, find, flatten} from "shared/utils"
import {getCodingDimensions} from "../../rater-rating/utils"
import {
  isParticipantFullyRatedByScenario,
  isParticipantRatedByQuestionnaire as checkIsParticipantRatedByQuestionnaire
} from "../utils"

export interface UseProjectModuleRatingHook {
  readonly loading: boolean
  readonly areAllProjectModulesRated: (surveyInvitation: SurveyInvitationLight) => boolean
}

export const useProjectModuleRating = (surveyId: UUID, useOnlyFinalScores = false): UseProjectModuleRatingHook => {
  const client = useApolloClient()

  const isMounted = React.useRef(false)
  const surveyRef = React.useRef<SurveyLight | null>(null)
  const projectModulesRef = React.useRef<ProjectModule[]>([])
  const codingItemsRef = React.useRef<CodingItem[]>([])
  const ratingsRef = React.useRef<Rating[]>([])

  const [codingDimensions, setCodingDimensions] = React.useState<CodingDimension[]>([])

  const {surveyLoading, survey} = useSurveyLight(surveyId)
  const {ratingsLoading: allRatingsLoading, ratings: allRatings} = useRatings(surveyId)
  const {projectModulesLoading, projectModules, getProjectModules} = useProjectModulesLazy()
  const {codingCriteriaLoading, codingCriteria, getCodingCriteria} = useCodingCriteriaByItemsList()
  const {
    freetextQuestionRatingsLoading,
    freetextQuestionRatings,
    getFreetextQuestionRatings
  } = useFreetextQuestionRatingsByRatingsList()
  const {
    scenarioCodingItemRatingsLoading,
    scenarioCodingItemRatings,
    getScenarioCodingItemRatings
  } = useScenarioCodingItemRatingsByRatingsList()

  const codingItems = getCodingItemsFromCodingDimensions(codingDimensions)

  const ratings = useOnlyFinalScores ? allRatings.filter(({isFinalScore}) => isFinalScore) : allRatings

  if (!isEqual(surveyRef.current, survey.orNull())) {
    surveyRef.current = survey.orNull()
  }

  if (!isEqual(projectModulesRef.current, projectModules)) {
    projectModulesRef.current = projectModules
  }

  if (!isEqual(codingItemsRef.current, codingItems)) {
    codingItemsRef.current = codingItems
  }

  if (!isEqual(ratingsRef.current, ratings)) {
    ratingsRef.current = ratings
  }

  const isParticipantRatedByQuestionnaire = (projectModule: ProjectModule, surveyInvitation: SurveyInvitationLight) =>
    checkIsParticipantRatedByQuestionnaire({
      projectModule,
      freetextQuestionRatingsForParticipant: freetextQuestionRatings.filter(
        freetextQuestionRating => freetextQuestionRating.surveyInvitationId === surveyInvitation.id
      ),
      ratings
    })

  const areAllProjectModulesRated = (surveyInvitation: SurveyInvitationLight) =>
    projectModules.length === 0 ||
    every(projectModule => {
      const projectModuleProgresses = find(
        moduleProgress =>
          projectModule.moduleType === ProjectModuleType.Scenario
            ? moduleProgress.scenarioId === projectModule.scenarioId
            : moduleProgress.questionnaireId === projectModule.questionnaireId,
        surveyInvitation.projectModuleProgresses
      )

      const hasParticipantCompletedProjectModule = projectModuleProgresses.exists(
        ({status}) => status === ProjectModuleProgressType.Completed
      )
      const wasProjectModuleRated =
        projectModule.moduleType === ProjectModuleType.Scenario
          ? isParticipantFullyRatedByScenario({
              projectModule,
              surveyInvitation,
              codingDimensions,
              codingItems,
              codingCriteria,
              scenarioCodingItemRatings,
              ratings
            })
          : isParticipantRatedByQuestionnaire(projectModule, surveyInvitation)

      return !hasParticipantCompletedProjectModule || wasProjectModuleRated
    }, projectModulesRef.current)

  React.useEffect(() => {
    const projectId = surveyRef.current?.projectId
    if (projectId !== undefined) {
      getProjectModules(projectId)
    }
  }, [surveyRef.current])

  React.useEffect(() => {
    const modules = projectModulesRef.current.filter(
      projectModule =>
        projectModule.moduleType === ProjectModuleType.Scenario &&
        projectModule.scenario !== null &&
        projectModule.scenario.codingModel !== null
    )
    Promise.all(modules.map(projectModule => getCodingDimensions(client, `${projectModule.scenario?.codingModel?.id}`)))
      .then(result => isMounted.current && setCodingDimensions(flatten(result)))
      .catch(() => isMounted.current && setCodingDimensions([]))
  }, [projectModulesRef.current])

  React.useEffect(() => {
    getCodingCriteria({items: codingItemsRef.current})
  }, [codingItemsRef.current])

  React.useEffect(() => {
    if (ratingsRef.current.length > 0) {
      getFreetextQuestionRatings(ratingsRef.current)
      getScenarioCodingItemRatings(ratingsRef.current)
    }
  }, [ratingsRef.current])

  React.useEffect(() => {
    isMounted.current = true
    return () => {
      isMounted.current = false
    }
  }, [])

  return {
    loading:
      surveyLoading ||
      projectModulesLoading ||
      codingCriteriaLoading ||
      scenarioCodingItemRatingsLoading ||
      allRatingsLoading ||
      freetextQuestionRatingsLoading,
    areAllProjectModulesRated
  }
}
