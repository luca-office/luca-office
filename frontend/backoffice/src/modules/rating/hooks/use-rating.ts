import {useRatings} from "shared/graphql/hooks"
import {Rating} from "shared/models"
import {find, Option} from "shared/utils"

export interface UseRatingHook {
  readonly rating: Option<Rating>
  readonly ratingLoading: boolean
}

export const useRating = (surveyId: UUID, userAccountId: UUID): UseRatingHook => {
  const {ratings, ratingsLoading} = useRatings(surveyId)

  const rating = find(rating => rating.userAccountId === userAccountId, ratings)

  return {rating, ratingLoading: ratingsLoading}
}
