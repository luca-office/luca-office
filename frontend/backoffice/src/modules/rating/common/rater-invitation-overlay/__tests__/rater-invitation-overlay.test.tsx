import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {Modal, Overlay} from "shared/components"
import {surveysMock, userAccountsMock} from "shared/graphql/__mocks__"
import {ManageInvitationsModal} from "../../../../common"
import {InviteAttendeesEmailAddressesGrid} from "../../../../common/manage-invitations-modal/invite-attendees-email-addresses-grid/invite-attendees-email-addresses-grid"
import {InviteAttendeesEmailTextarea} from "../../../../common/manage-invitations-modal/invite-attendees-email-textarea/invite-attendees-email-textarea"
import * as useRaterInvitationOverlayHook from "../hooks/use-rater-invitation-overlay"
import {UseRaterInvitationOverlayHook} from "../hooks/use-rater-invitation-overlay"
import {RaterInvitationOverlay, RaterInvitationOverlayProps} from "../rater-invitation-overlay"

const survey = surveysMock[0]
const emails = userAccountsMock.map(({email}) => email)

const defaultProps: RaterInvitationOverlayProps = {
  surveyId: survey.id,
  existingRaterEmails: emails,
  onDismiss: jest.fn()
}

const stateHookValuesDefault: UseRaterInvitationOverlayHook = {
  dataLoading: false,
  actionLoading: false,
  inviteRaters: jest.fn(() => Promise.resolve()),
  setEmails: jest.fn(),
  showRaterWarning: false
}

const stateSpy = jest.spyOn(useRaterInvitationOverlayHook, "useRaterInvitationOverlay")

const getComponent = (props?: Partial<RaterInvitationOverlayProps>) => (
  <RaterInvitationOverlay {...{...defaultProps, ...props}} />
)

describe("rater-invitation-overlay", () => {
  it("renders correctly", async () => {
    stateSpy.mockReturnValue(stateHookValuesDefault)
    const component = create(getComponent())

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("has correct structure", async () => {
    stateSpy.mockReturnValue(stateHookValuesDefault)
    const component = shallow(getComponent())

    const overlay = component.find(Overlay)
    expect(overlay).toHaveLength(1)

    const invitationsModal = component.find(Overlay).dive().find(ManageInvitationsModal)
    expect(invitationsModal).toHaveLength(1)

    const modal = invitationsModal.dive().find(Modal)
    expect(modal).toHaveLength(1)

    const modalContent = modal.dive()
    expect(modalContent.find(InviteAttendeesEmailTextarea)).toHaveLength(1)
    expect(modalContent.find(InviteAttendeesEmailAddressesGrid)).toHaveLength(1)
  })
  it("triggers dismiss correctly", async () => {
    const onDismissMock = jest.fn()
    stateSpy.mockReturnValue(stateHookValuesDefault)
    const component = shallow(getComponent({onDismiss: onDismissMock}))

    const invitationsModal = component.find(Overlay).dive().find(ManageInvitationsModal)
    expect(invitationsModal).toHaveLength(1)

    const modal = invitationsModal.dive().find(Modal)
    expect(modal).toHaveLength(1)

    const onDismissHandler = modal.prop("onDismiss")
    expect(onDismissHandler).toBeDefined()

    onDismissHandler!()
    expect(onDismissMock).toHaveBeenCalledTimes(1)
  })
})
