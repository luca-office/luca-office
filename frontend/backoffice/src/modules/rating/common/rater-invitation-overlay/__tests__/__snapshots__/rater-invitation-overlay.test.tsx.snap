// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rater-invitation-overlay renders correctly 1`] = `
<div
  className="css-1iw6hgd-Overlay"
>
  <div
    className="css-1iw6hgd-Overlay"
  >
    <div>
      <div
        className="css-osbzd6-Modal"
      >
        <header
          className="css-14sl779-Modal"
        >
          <div
            className="css-1f6p2vj-Heading"
          >
            rating__invite_title
          </div>
          <div
            className="css-12vbzj8-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[MockFunction]}
          />
        </header>
        <section
          className="css-1gs9x9w-Modal"
        >
          <div
            className="wrapper css-kv7sb5-ManageInvitationsModal"
          >
            <div>
              <div
                className="css-duzsc9-Paper"
              >
                <div
                  className="css-1dn2d64-Heading"
                >
                  projects__survey_details_invite_emails_heading
                </div>
                <div
                  className="css-1p9qd62-Text"
                >
                  rating__invite_description
                </div>
                <div
                  className="css-prjr7p"
                >
                  <textarea
                    className="css-7iyx5a"
                    disabled={false}
                    onChange={[Function]}
                    placeholder="projects__survey_details_invite_emails_placeholder"
                    rows={7}
                    value=""
                  />
                </div>
                <div
                  className="css-v6owty-Text"
                >
                  projects__survey_details_invite_emails_number_of_valid_addresses
                </div>
              </div>
            </div>
            <div
              className="wrapper css-z70zj5-InviteAttendeesEmailAddressesGrid"
            >
              <div
                className="css-1akphf9-Heading"
              >
                rating__invite_emails_title
              </div>
              <div
                className="grid-wrapper css-1d467hy-InviteAttendeesEmailAddressesGrid"
              >
                <div
                  className="grid-top-bar css-7h2thf-InviteAttendeesEmailAddressesGrid"
                />
                <div
                  className="grid-scroll-wrapper css-v0ses7-InviteAttendeesEmailAddressesGrid"
                >
                  <div
                    className="grid css-b7vqx3-InviteAttendeesEmailAddressesGrid"
                  >
                    <div
                      className="css-0"
                      id="tooltip_42"
                    >
                      <div>
                        <div
                          className="css-bn1a4g-Paper"
                          title="projects__survey_details_tooltip_invite_already_sent"
                        >
                          <div
                            className="css-17zbm7y-Text"
                          >
                            test-user@cap3.de
                          </div>
                          <div
                            className="css-0"
                            id="tooltip_42"
                          >
                            <div
                              className="trash-icon css-16kls7j-Icon"
                              dangerouslySetInnerHTML={
                                {
                                  "__html": {},
                                }
                              }
                              onClick={[Function]}
                              title="projects__survey_details_tooltip_remove_from_list"
                            />
                          </div>
                          <div
                            className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                            role="tooltip"
                            style={
                              {
                                "backgroundColor": "#2e3032",
                                "zIndex": 1,
                              }
                            }
                          >
                            projects__survey_details_tooltip_remove_from_list
                            <div
                              className="react-tooltip-arrow styles-module_arrow__K0L3T"
                              style={{}}
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                      role="tooltip"
                      style={
                        {
                          "backgroundColor": "#2e3032",
                          "zIndex": 1,
                        }
                      }
                    >
                      projects__survey_details_tooltip_invite_already_sent
                      <div
                        className="react-tooltip-arrow styles-module_arrow__K0L3T"
                        style={{}}
                      />
                    </div>
                    <div
                      className="css-0"
                      id="tooltip_42"
                    >
                      <div>
                        <div
                          className="css-bn1a4g-Paper"
                          title="projects__survey_details_tooltip_invite_already_sent"
                        >
                          <div
                            className="css-17zbm7y-Text"
                          >
                            max-mustermann@cap3.de
                          </div>
                          <div
                            className="css-0"
                            id="tooltip_42"
                          >
                            <div
                              className="trash-icon css-16kls7j-Icon"
                              dangerouslySetInnerHTML={
                                {
                                  "__html": {},
                                }
                              }
                              onClick={[Function]}
                              title="projects__survey_details_tooltip_remove_from_list"
                            />
                          </div>
                          <div
                            className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                            role="tooltip"
                            style={
                              {
                                "backgroundColor": "#2e3032",
                                "zIndex": 1,
                              }
                            }
                          >
                            projects__survey_details_tooltip_remove_from_list
                            <div
                              className="react-tooltip-arrow styles-module_arrow__K0L3T"
                              style={{}}
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                      role="tooltip"
                      style={
                        {
                          "backgroundColor": "#2e3032",
                          "zIndex": 1,
                        }
                      }
                    >
                      projects__survey_details_tooltip_invite_already_sent
                      <div
                        className="react-tooltip-arrow styles-module_arrow__K0L3T"
                        style={{}}
                      />
                    </div>
                    <div
                      className="css-0"
                      id="tooltip_42"
                    >
                      <div>
                        <div
                          className="css-bn1a4g-Paper"
                          title="projects__survey_details_tooltip_invite_already_sent"
                        >
                          <div
                            className="css-17zbm7y-Text"
                          >
                            marianne-musterfrau@cap3.de
                          </div>
                          <div
                            className="css-0"
                            id="tooltip_42"
                          >
                            <div
                              className="trash-icon css-16kls7j-Icon"
                              dangerouslySetInnerHTML={
                                {
                                  "__html": {},
                                }
                              }
                              onClick={[Function]}
                              title="projects__survey_details_tooltip_remove_from_list"
                            />
                          </div>
                          <div
                            className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                            role="tooltip"
                            style={
                              {
                                "backgroundColor": "#2e3032",
                                "zIndex": 1,
                              }
                            }
                          >
                            projects__survey_details_tooltip_remove_from_list
                            <div
                              className="react-tooltip-arrow styles-module_arrow__K0L3T"
                              style={{}}
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                      role="tooltip"
                      style={
                        {
                          "backgroundColor": "#2e3032",
                          "zIndex": 1,
                        }
                      }
                    >
                      projects__survey_details_tooltip_invite_already_sent
                      <div
                        className="react-tooltip-arrow styles-module_arrow__K0L3T"
                        style={{}}
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="grid-bottom-bar css-121255i-InviteAttendeesEmailAddressesGrid"
                />
              </div>
            </div>
          </div>
        </section>
        <footer
          className="css-ukdqb3-Modal"
        >
          <button
            className="cancel-button css-ek72a-Button"
            disabled={false}
            onClick={[Function]}
          >
            <div
              className="css-95j2hm-Button"
            >
              cancel_button
            </div>
          </button>
          <button
            className="confirm-button css-o1ky6y-Button"
            disabled={true}
            onClick={[Function]}
          >
            <div
              className="css-95j2hm-Button"
            >
              rating__invite_title
            </div>
          </button>
        </footer>
      </div>
    </div>
  </div>
</div>
`;
